tests/contrib/operators/test_gcp_compute_operator.py

```diff
@@ -0,0 +1,242 @@
+            project_id=PROJECT_ID,
+            zone=LOCATION,
+            resource_id=RESOURCE_ID,
+            body=BODY,
+            task_id='id'
+        )
+        result = op.execute(None)
+        mock_hook.assert_called_once_with(api_version='v1',
+                                          gcp_conn_id='google_cloud_default')
+        mock_hook.return_value.set_machine_type.assert_called_once_with(
+            PROJECT_ID, LOCATION, RESOURCE_ID, BODY
+        )
+        self.assertTrue(result)
+
+    @mock.patch('airflow.contrib.operators.gcp_compute_operator.GceHook')
+    def test_set_machine_type_should_throw_ex_when_missing_project_id(self, mock_hook):
+        with self.assertRaises(AirflowException) as cm:
+            op = GceSetMachineTypeOperator(
+                project_id="",
+                zone=LOCATION,
+                resource_id=RESOURCE_ID,
+                body=BODY,
+                task_id='id'
+            )
+            op.execute(None)
+        err = cm.exception
+        self.assertIn("The required parameter 'project_id' is missing", str(err))
+        mock_hook.assert_not_called()
+
+    @mock.patch('airflow.contrib.operators.gcp_compute_operator.GceHook')
+    def test_set_machine_type_should_throw_ex_when_missing_zone(self, mock_hook):
+        with self.assertRaises(AirflowException) as cm:
+            op = GceSetMachineTypeOperator(
+                project_id=PROJECT_ID,
+                zone="",
+                resource_id=RESOURCE_ID,
+                body=BODY,
+                task_id='id'
+            )
+            op.execute(None)
+        err = cm.exception
+        self.assertIn("The required parameter 'zone' is missing", str(err))
+        mock_hook.assert_not_called()
+
+    @mock.patch('airflow.contrib.operators.gcp_compute_operator.GceHook')
+    def test_set_machine_type_should_throw_ex_when_missing_resource_id(self, mock_hook):
+        with self.assertRaises(AirflowException) as cm:
+            op = GceSetMachineTypeOperator(
+                project_id=PROJECT_ID,
+                zone=LOCATION,
+                resource_id="",
+                body=BODY,
+                task_id='id'
+            )
+            op.execute(None)
+        err = cm.exception
+        self.assertIn("The required parameter 'resource_id' is missing", str(err))
+        mock_hook.assert_not_called()
+
+    @mock.patch('airflow.contrib.operators.gcp_compute_operator.GceHook')
+    def test_set_machine_type_should_throw_ex_when_missing_machine_type(self, mock_hook):
+        with self.assertRaises(AirflowException) as cm:
+            op = GceSetMachineTypeOperator(
+                project_id=PROJECT_ID,
+                zone=LOCATION,
+                resource_id=RESOURCE_ID,
+                body={},
+                task_id='id'
+            )
+            op.execute(None)
+        err = cm.exception
+        self.assertIn(
+            "The required body field '{}' is missing. Please add it.".format(
```

Comment:
I prefer to keep it as string without formatting. In the future if we change any wording, it will be as simple as copy-pasting the actual message from unit test error.

Reply:
True, but on the other hand this is the way the string is in the `FieldValidator`. So one could say if we change it in the `FieldValidator` it would be easier to copy paste from there ;) 
But you're probably right and it also looks weird here like this, so I'll change it.

Reply:
Done in 585a1f0